### ISSUE TITLE
test failure on 64-bit Ubuntu 12.04

### ISSUE THREAD

**Issue by staticfloat** [JuliaAttic/libuv-archive#8]
I'm trying to package libuv on Ubuntu, and I'm getting the following errors when running the tests:

```
/home/sabae/src/libuv-julia0.2/run-tests
[%   0|+   0|-   0]: platform_output
Output from process `platform_output`:
uv_get_process_title: /home/sabae/src/libuv-julia0.2/run-tests
uv_resident_set_memory: 741376
uv_uptime: 209692.239093
uv_cpu_info:
  model: Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz
  speed: 3377
  times.sys: 2886500
  times.user: 25381300
  times.idle: 2063898600
  times.irq: 0
  times.nice: 177400
uv_interface_addresses:
  name: lo
  internal: 1
  address: 127.0.0.1
  name: eth0
  internal: 0
  address: 10.0.2.15
  name: lo
  internal: 1
  address: ::1
  name: eth0
  internal: 0
  address: fe80::a00:27ff:fe49:bc31
=============================================================
[%  10|+  19|-   0]: ipc_listen_before_write
`ipc_listen_before_write` failed: exit code 134
Output from process `ipc_listen_before_write`:
exit_cb
Assertion failed in test/test-ipc.c on line 338: local_conn_accepted == 1
=============================================================
[%  10|+  19|-   1]: ipc_listen_after_write
`ipc_listen_after_write` failed: exit code 134
Output from process `ipc_listen_after_write`:
exit_cb
Assertion failed in test/test-ipc.c on line 348: local_conn_accepted == 1
=============================================================
[%  11|+  19|-   2]: ipc_send_recv_pipe
`ipc_send_recv_pipe` failed: exit code 134
Output from process `ipc_send_recv_pipe`:
Assertion failed in test/test-ipc-send-recv.c on line 66: pending == ctx.expected_type
=============================================================
[%  12|+  19|-   3]: ipc_send_recv_tcp
`ipc_send_recv_tcp` failed: exit code 134
Output from process `ipc_send_recv_tcp`:
Assertion failed in test/test-ipc-send-recv.c on line 66: pending == ctx.expected_type
=============================================================
[%  12|+  19|-   4]: ipc_tcp_connection
`ipc_tcp_connection` failed: exit code 134
Output from process `ipc_tcp_connection`:
exit_cb
Assertion failed in test/test-ipc.c on line 358: read2_cb_called == 1
=============================================================
[% 100|+ 178|-   5]: Done.
make[1]: *** [test] Error 5
```

These errors do not appear on the `master` branch, only `julia0.2`

**Comment by vtjnash** [comment]
I didn't fix those tests, the build should be ok anyways

**Comment by staticfloat** [comment]
for now, I will disable these tests, however it is unlikely this solution will be palatable to anyone looking to package this in an official distribution.  The best course of action is, of course, to take the time to fix the tests, although completely removing the tests (if they are no longer applicable, etc...) is also a possible solution

**Comment by vtjnash** [comment]
they're still applicable, it just a matter of fixing a few constants in the tests (the last argument to the pipe constructor needs the direction specified). i didn't do it simply because it seemed that upstream is unlikely to merge our work directly and will choose to rewrite it anyways. Since we were building a private static library -- currently incompatible with upstream in a few important places -- I didn't think anyone would notice :)

**Comment by staticfloat** [comment]
Ah, I see.  This seems like a strange thing to package all around then (even though it looks like upstream itself isn't widely packaged). "Private static library" isn't usually compatible with packaging, but since this is mostly just for Travis, I think we can bend the rules.

Do you think this is something you will eventually get around to?  If possible, I'd like to keep the testing in the debian packaging process.  I took a look at it, and as I am 100% not familiar with libuv, it's difficult for me to tell what's going on.  If you don't have time however, I can invest the time to learn a bit. :)

**Comment by vtjnash** [comment]
You can fix the tests by finding any line that initializes a pipe (calls uv_pipe_init) and replace the third argument with `UV_PIPE_READABLE|UV_PIPE_WRITEABLE` if it was 0 and `UV_PIPE_READABLE|UV_PIPE_WRITEABLE|UV_PIPE_IPC` if it was 1

Or you can probably get the same effect by uncommenting the compatibility lines in uv_pipe_init that simulate that fix.

I opened an issue with libuv for the underlying issue that caused that breakage: https://github.com/joyent/libuv/issues/773

Yes, I've been recommending against packaging it for that reason. But node/libuv is starting to see enough use it will start to be more common to see packages of it, I suspect. For our purposes, there is a rather significant bug in the process spawning code, which had no effect on libuv, but forced us to create a fork. (our fix breaks an important feature for node, so it isn't trivial to merge, until they get around to refactoring their IPC api)

**Comment by staticfloat** [comment]
Closed by #10
